repl/src/main/resources/fake_shell.py

```diff
@@ -83,6 +91,79 @@ def execute_reply_internal_error(message, exc_info=None):
     })
 
 
+class JobContextImpl(object):
+    def __init__(self):
+        self.lock = threading.Lock()
+        self.sc = global_dict['sc']
+        self.sql_ctx = global_dict['sqlContext']
+        self.hive_ctx = None
+        self.streaming_ctx = None
+        self.local_tmp_dir_path = local_tmp_dir_path
+
+    def sc(self):
+        return self.sc
+
+    def sql_ctx(self):
+        return self.sql_ctx
+
+    def hive_ctx(self):
+        if self.hive_ctx is None:
```

Comment:
You could check if self.sql_ctx is an instance of `HiveContext` and use that if it is.